File: cokemania2/LTC/100_same_tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        def check_together(p, q):
            if p == None and q == None:
                return True
            elif p and q and p.val == q.val:
                return check_together(p.left, q.left) and check_together(p.right, q.right)
            return False
    
        return check_together(p, q)

File: hall_of_fame/jaewon/100_sameTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> v;
    void getNodes(TreeNode* tn){
        if(tn!=nullptr){
            v.push_back(tn->val);
            getNodes(tn->left);
            getNodes(tn->right);
        }else{
            v.push_back(-1);
        }
    }
    
    bool isSameTree(TreeNode* p, TreeNode* q) {
        getNodes(p);
        getNodes(q);
        if(v.size()%2!=0){
            return false;
        }else{
            for(int i=0; i<v.size()/2; i++){
                if(v[i] != v[i+v.size()/2]){
                    return false;
                }
            }
            return true;
        }
    }
};
